src/sentry/utils/geo.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import absolute_import
+
+import logging
+
+from django.conf import settings
+
+
+logger = logging.getLogger(__name__)
+
+
+# default is no-op
+geo_by_addr = lambda ip: None
+
+try:
+    import GeoIP
+except ImportError:
+    logger.warning("GeoIP module not available.")
+else:
+    geoip_path = getattr(settings, 'GEOIP_PATH', None)
+    if geoip_path:
+        try:
+            geo_db = GeoIP.open(geoip_path, GeoIP.GEOIP_STANDARD)
+        except BaseException:
+            logger.warning("Error opening GeoIP database: %s" % geoip_path)
+        else:
+            def geo_by_addr(ip):
+                try:
+                    return geo_db.record_by_addr(ip)
+                except BaseException:
```

Comment:
Not down with this `BaseException`. Any reason not just `Exeception`? Or simpler yet, just do like, `geo_by_addr = geo_db.record_by_addr` and let the caller handler it.